controlplane/kubeadm/internal/controllers/helpers.go

```diff
@@ -102,20 +98,39 @@ func (r *KubeadmControlPlaneReconciler) reconcileKubeconfig(ctx context.Context,
 	return ctrl.Result{}, nil
 }
 
-func (r *KubeadmControlPlaneReconciler) adoptKubeconfigSecret(ctx context.Context, cluster *clusterv1.Cluster, configSecret *corev1.Secret, controllerOwnerRef metav1.OwnerReference) error {
+// Add the controlplane ownerReference if it's not present.
+// If the kubeconfig secret was created by v1alpha2 controllers, and thus it has the Cluster as the owner instead of KCP;
```

Comment:
I think the comment is slightly outdated.

Seems like we always set KCP as the controller owner ref if necessary + 
remove previous controller ownerRef (if it's there) + 
remove cluster ownerRef (if it's there)

Reply:
(just in case you missed it, seems still open)